### DISCUSSION TITLE
The issue #504 - Shift key is pressed.. affecting Ubuntu 24.04 (Guidance on how to proceed)

### DISCUSSION THREAD

**Discussion by new-creation7** [mike-fabian/ibus-typing-booster#548] [General]
Greetings,
Because of the [bug](https://github.com/mike-fabian/ibus-typing-booster/issues/504) (The shift key is registered as pressed) there is weird behaviors across apps (like libreoffice writer).
The issue is solved in Release 2.25.6 (Thanks)

But the version of ibus-typing-booster is currently Version: 2.25.4-1 on Ubuntu 24.04 repositories.

Now how to inform Ubuntu so that they will upgrade the package (as all Ubuntu users benefit from it)? 

Thanks

**Comment by mike-fabian** [comment]
@new-creation7 

If I  look here: https://repology.org/project/ibus-typing-booster/versions
it looks like 2.25.18 packages exist for Debian 13 and Debian unstable.

You could probably install these on Ubuntu 24.04 and they should work, is that right?

Would that help already? 

I have no idea how to inform Ubuntu to do the  upgrade.

**Reply by mike-fabian** [reply]
Does this help?: https://github.com/mike-fabian/ibus-typing-booster/discussions/713

**Comment by mike-fabian** [comment]
There still seems to be no update for Ubuntu 24.04.

But another user recently ran into the same problem:
https://github.com/mike-fabian/ibus-typing-booster/issues/762
It took me a while to realize that it was the same problem, but indeed it was. Please look at this comment:

https://github.com/mike-fabian/ibus-typing-booster/issues/762#issuecomment-3018394911

Even though there is no updated package for Ubuntu 24.04, one can easily fix the problem
by installing the latest package from Debian unstable, it installs and works without problem:

```
mfabian@ubuntu-24:~$ wget http://ftp.us.debian.org/debian/pool/main/i/ibus-typing-booster/ibus-typing-booster_2.27.46-1_all.deb
...

mfabian@ubuntu-24:~$ sudo apt install ./ibus-typing-booster_2.27.46-1_all.deb
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'ibus-typing-booster' instead of './ibus-typing-booster_2.27.46-1_all.deb'
The following additional packages will be installed:
  python3-rapidfuzz
The following NEW packages will be installed:
  python3-rapidfuzz
The following packages will be upgraded:
  ibus-typing-booster
1 upgraded, 1 newly installed, 0 to remove and 1 not upgraded.
Need to get 2,021 kB/14.7 MB of archives.
After this operation, 13.2 MB of additional disk space will be used.
Do you want to continue? [Y/n] y
Get:1 /home/mfabian/ibus-typing-booster_2.27.46-1_all.deb ibus-typing-booster all 2.27.46-1 [12.7 MB]
Get:2 http://de.archive.ubuntu.com/ubuntu noble/universe amd64 python3-rapidfuzz amd64 3.6.2+ds-1 [2,021 kB]
Fetched 2,021 kB in 0s (7,528 kB/s)          
(Reading database ... 175149 files and directories currently installed.)
Preparing to unpack .../ibus-typing-booster_2.27.46-1_all.deb ...
Unpacking ibus-typing-booster (2.27.46-1) over (2.25.4-1) ...
Selecting previously unselected package python3-rapidfuzz.
Preparing to unpack .../python3-rapidfuzz_3.6.2+ds-1_amd64.deb ...
Unpacking python3-rapidfuzz (3.6.2+ds-1) ...
Setting up ibus-typing-booster (2.27.46-1) ...
Setting up python3-rapidfuzz (3.6.2+ds-1) ...
Processing triggers for hicolor-icon-theme (0.17-2) ...
Processing triggers for gnome-menus (3.36.0-1.1ubuntu3) ...
Processing triggers for libglib2.0-0t64:amd64 (2.80.0-6ubuntu3.4) ...
Processing triggers for desktop-file-utils (0.27-2build1) ...
N: Download is performed unsandboxed as root as file '/home/mfabian/ibus-typing-booster_2.27.46-1_all.deb' couldn't be accessed by user '_apt'. - pkgAcquire::Run (13: Permission denied)
mfabian@ubuntu-24:~$ 
```
And then `ibus restart` and the problem is fixed.
